src/Ember/Client/index.ts

```diff
@@ -388,24 +394,24 @@ export class EmberClient extends EventEmitter<EmberClientEvents> {
 			const i = pathArr.shift()
 			if (!i) break // TODO - this will break the loop if the path was `1..0`
 			if (!tree) break
-			let next = getNextChild(tree, i)
-			if (!next) {
-				const req = await this.getDirectory(tree)
-				tree = (await req.response) as NumberedTreeNode<EmberElement>
-				next = getNextChild(tree, i)
-			}
-			tree = next
+			const req = await this.getDirectory(tree)
+			tree = (await req.response) as NumberedTreeNode<EmberElement>
+			tree = getNextChild(tree, i)
 			if (!tree) throw new Error(`Could not find node ${i} on given path ${numberedPath.join()}`)
 			if (tree?.number !== undefined) numberedPath.push(tree.number)
 		}
-
-		if (cb && numberedPath) {
-			this._subscriptions.push({
-				path: numberedPath.join('.'),
-				cb,
-			})
+		if (tree) {
+			const req = await this.getDirectory(tree)
 		}
 
+		// this resulted in doubled subscription because getDirectory() adds the element as well
```

Comment:
I don't understand what's going on here. Since the callback is never passed to the getDirectory call it should not cause a double subscription.